src/StreamJsonRpc/Reflection/RpcMarshalableKnownSubtypeAttribute.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc;
+
+/// <summary>
+/// Attribute to be used on an interface marked with <see cref="RpcMarshalableAttribute"/> to indicate that a known
+/// inherited interface exists.
+///
+/// When an object that implements <see cref="SubType"/> is marshaled as its base interface marked with
+/// <see cref="RpcMarshalableKnownSubtypeAttribute"/>, <see cref="SubTypeCode"/> is included in the StreamJsonRpc
+/// message allowing the creation of a proxy which implements <see cref="SubType"/>.
+/// </summary>
+/// <remarks>
+/// If an object implements more than one of the <see cref="SubType"/> insterfaces, the behavior is undefined.
```

Comment:
Do we want to be more precise? Use the first matching or last matching (by SubTypeCode ordering)

Reply:
Ya, undefined is probably unacceptable, as it could mean things work and then break later. At the very least, we should explicitly check for this condition and force a failure. 
If we want to go beyond that and support multiple interfaces, that's fine. Maybe when we encode the subtype id's, we can do so as an array instead of a scalar so we can signify all the interfaces that are allowed.